### PR TITLE
Add support for X-Databricks-User-Agent header

### PR DESCRIPTION
**Pull request by antas-marcin**
### What's being changed:

Add support for X-Databricks-User-Agent header in all Databricks modules.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: modules/generative-databricks/clients/databricks.go [hunk 1]
```diff
@@ -279,6 +282,10 @@ func (v *databricks) getApiKeyFromContext(ctx context.Context, apiKey, envVarVal
 	return "", fmt.Errorf("no api key found neither in request header: %s nor in environment variable under %s", apiKey, envVar)
 }
 
+func (v *databricks) getHeaderValue(ctx context.Context, header string) string {
```
Comment (dirkkul) [inline modules/generative-databricks/clients/databricks.go:285]: why add this function instead of using `modulecomponents.GetValueFromContext` directly?

Comment (antas-marcin) [reply modules/generative-databricks/clients/databricks.go:285]: that's my style of writing :) can change it if you want :)

Comment (dirkkul) [reply modules/generative-databricks/clients/databricks.go:285]: As the function does nothing besides calling modulecomponents.GetValueFromContext I'd prefer if you'd remove `getHeaderValue`. We have too much copy&paste in the modules and the more is shared the better :)

Comment (antas-marcin) [reply modules/generative-databricks/clients/databricks.go]: fixed!

**Review by dirkkul — APPROVED** [review]

**Pull request merged**